Mathlib/Tactic/Relation/Rfl.lean

```diff
@@ -25,6 +25,22 @@ relation, that is, a relation which has a reflexive lemma tagged with the attrib
 def rflTac : TacticM Unit :=
   withMainContext do liftMetaFinishingTactic (·.applyRfl)
 
+/-- If the goal is the form `x ~ y`, where `~` is a reflexive
+relation, return `some ((· ~ ·), x, y)`. -/
+def _root_.Lean.Expr.isReflRel (e : Expr) : MetaM (Option (Name × Expr × Expr)) := do
+  if let some (_, lhs, rhs) := e.eq? then
+    return (``Eq, lhs, rhs)
+  if let some (lhs, rhs) := e.iff? then
+    return (``Iff, lhs, rhs)
+  if let some (_, lhs, _, rhs) := e.heq? then
+    return (``HEq, lhs, rhs)
+  if let .app (.app rel lhs) rhs := e then
+    unless (← (reflExt.getState (← getEnv)).getMatch rel reflExt.config).isEmpty do
+      match rel.getAppFn.constName? with
```

Comment:
If you're intending to use `isReflRel` with `mkRel`, then there's a bug because there's no reason to assume that this constant takes only implicit arguments followed by two explicit arguments.

We could fix `mkRel` so that it uses `mkAppOptM` and feeds in `lhs` and `rhs` as the last two explicit arguments. Or, CC could document this limitation with its ability to handle reflexive/symmetric relations and we can fix it later.

Name-wise, I think it shouldn't use `is` since it doesn't return a `Bool`, and it should end in `?` since it returns an `Option`. Perhaps `Lean.Expr.relSidesIfRefl?`, though I'm not really sure.